crates/clvm-traits/src/match_byte.rs

```diff
@@ -5,19 +5,14 @@ use crate::{ClvmDecoder, ClvmEncoder, FromClvm, FromClvmError, ToClvm, ToClvmErr
 #[derive(Debug, Copy, Clone)]
 pub struct MatchByte<const BYTE: u8>;
 
-impl<N, const BYTE: u8> ToClvm<N> for MatchByte<BYTE> {
-    fn to_clvm(&self, encoder: &mut impl ClvmEncoder<Node = N>) -> Result<N, ToClvmError> {
-        if BYTE == 0 {
-            return encoder.encode_atom(&[]);
-        }
-        let number = BigInt::from(BYTE);
-        let bytes = number.to_signed_bytes_be();
```

Comment:
the old code here looks a bit suspicious too, it doesn't handle leading zeroes based on whether the number of positive or negative at all. But you're changing this behavior now, afaict.

Reply:
Fixed for decoding and added a test for each byte value with a macro